hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/codec/ReconOMDBDefinition.java

```diff
@@ -0,0 +1,405 @@
+
+  /** openFileTable: /volumeId/bucketId/parentId/fileName/id :- KeyInfo. */
+  public static final DBColumnFamilyDefinition<String, OmKeyInfo> OPEN_FILE_TABLE_DEF
+      = new DBColumnFamilyDefinition<>(OPEN_FILE_TABLE, StringCodec.get(), CUSTOM_CODEC_FOR_KEY_TABLE);
+
+  public static final Codec<OmDirectoryInfo> CUSTOM_CODEC_FOR_DIR_TABLE = new DelegatedCodec<>(
+      Proto2Codec.get(OzoneManagerProtocolProtos.DirectoryInfo.getDefaultInstance()),
+      ReconOMDBDefinition::getOmDirInfoFromProtobuf,
+      null,
+      OmDirectoryInfo.class);
+  /** directoryTable: /volumeId/bucketId/parentId/dirName :- DirInfo. */
+  public static final DBColumnFamilyDefinition<String, OmDirectoryInfo> DIRECTORY_TABLE_DEF
+      = new DBColumnFamilyDefinition<>(DIRECTORY_TABLE, StringCodec.get(), CUSTOM_CODEC_FOR_DIR_TABLE);
+
+  /** deletedDirectoryTable: /volumeId/bucketId/parentId/dirName/objectId :- KeyInfo. */
+  public static final DBColumnFamilyDefinition<String, OmKeyInfo> DELETED_DIR_TABLE_DEF
+      = new DBColumnFamilyDefinition<>(DELETED_DIR_TABLE, StringCodec.get(), CUSTOM_CODEC_FOR_KEY_TABLE);
+
+  //---------------------------------------------------------------------------
+  public static final Map<String, DBColumnFamilyDefinition<?, ?>> COLUMN_FAMILIES
+      = DBColumnFamilyDefinition.newUnmodifiableMap(
+      BUCKET_TABLE_DEF,
+      DELETED_DIR_TABLE_DEF,
+      DELETED_TABLE_DEF,
+      DIRECTORY_TABLE_DEF,
+      DELEGATION_TOKEN_TABLE_DEF,
+      FILE_TABLE_DEF,
+      KEY_TABLE_DEF,
+      META_TABLE_DEF,
+      MULTIPART_INFO_TABLE_DEF,
+      OPEN_FILE_TABLE_DEF,
+      OPEN_KEY_TABLE_DEF,
+      PREFIX_TABLE_DEF,
+      PRINCIPAL_TO_ACCESS_IDS_TABLE_DEF,
+      S3_SECRET_TABLE_DEF,
+      SNAPSHOT_INFO_TABLE_DEF,
+      SNAPSHOT_RENAMED_TABLE_DEF,
+      COMPACTION_LOG_TABLE_DEF,
+      TENANT_ACCESS_ID_TABLE_DEF,
+      TENANT_STATE_TABLE_DEF,
+      TRANSACTION_INFO_TABLE_DEF,
+      USER_TABLE_DEF,
+      VOLUME_TABLE_DEF);
+
+  private static final ReconOMDBDefinition INSTANCE = new ReconOMDBDefinition();
+
+  /**
+   * Parses BucketInfo protobuf and creates OmBucketInfo without deserializing ACL list.
+   * @param bucketInfo
+   * @return instance of OmBucketInfo
+   */
+  public static OmBucketInfo getOmBucketInfoFromProtobuf(OzoneManagerProtocolProtos.BucketInfo bucketInfo) {
+    OmBucketInfo.Builder obib = OmBucketInfo.newBuilder()
+        .setVolumeName(bucketInfo.getVolumeName())
+        .setBucketName(bucketInfo.getBucketName())
+        .setIsVersionEnabled(bucketInfo.getIsVersionEnabled())
+        .setStorageType(StorageType.valueOf(bucketInfo.getStorageType()))
+        .setCreationTime(bucketInfo.getCreationTime())
+        .setUsedBytes(bucketInfo.getUsedBytes())
+        .setModificationTime(bucketInfo.getModificationTime())
+        .setQuotaInBytes(bucketInfo.getQuotaInBytes())
+        .setUsedNamespace(bucketInfo.getUsedNamespace())
+        .setQuotaInNamespace(bucketInfo.getQuotaInNamespace());
```

Comment:
Instead of re-implementing most of the conversion logic without ACL, please:
- move this to `OmBucketInfo` as a helper that returns `OmBucketInfo.Builder`
- let both "full" and "reduced" conversion call the helper and then `build()`, with "full" adding further logic between the two steps

Apply the same to `OmKeyInfo`, etc.

Reply:
Pls review again. I have tried to refactor by reusing as much as possible.